src/GRA.Data/Repository/PsBranchSelectionRepository.cs

```diff
@@ -32,15 +32,15 @@ public async Task<int> GetCountByKitIdAsync(int kitId)
         {
             return await DbSet
                 .AsNoTracking()
-                .Where(_ => _.KitId.HasValue && _.KitId == kitId)
+                .Where(_ => _.KitId.HasValue && _.KitId == kitId && !_.IsDeleted)
```

Comment:
Move the Where clause into the `CountAsync()`

Reply:
7ea0e75f2f09c9275a89a5a37a81ef42cf2b2fba